contrib/valkey-go/valkey.go

```diff
@@ -0,0 +1,369 @@
+	coreClient
+	dedicatedClient valkey.DedicatedClient
+}
+
+func NewClient(option valkey.ClientOption, opts ...ClientOption) (valkey.Client, error) {
+	valkeyClient, err := valkey.NewClient(option)
+	if err != nil {
+		return nil, err
+	}
+	var cfg clientConfig
+	defaults(&cfg)
+	for _, fn := range opts {
+		fn(&cfg)
+	}
+	var host string
+	var port int
+	if len(option.InitAddress) == 1 {
+		host, port = splitHostPort(option.InitAddress[0])
+	}
+	core := coreClient{
+		Client:       valkeyClient,
+		option:       option,
+		clientConfig: cfg,
+		spanName:     namingschema.OpName(namingschema.ValkeyOutbound),
+		host:         host,
+		port:         port,
+	}
+	return &client{
+		coreClient: core,
+	}, nil
+}
+
+func splitHostPort(addr string) (string, int) {
+	host, portStr, err := net.SplitHostPort(addr)
+	if err != nil {
+		log.Error("%q cannot be split: %s", addr, err)
+		return "", 0
+	}
+	port, _ := strconv.Atoi(portStr)
+	return host, port
+}
+
+type commander interface {
+	Commands() []string
+}
+
+func processCmd(commander commander) (command, statement string, size int) {
+	commands := commander.Commands()
+	if len(commands) == 0 {
+		return "", "", 0
+	}
+	command = commands[0]
+	statement = strings.Join(commands, "\n")
+	return command, statement, len(statement)
+}
+
+func processMultiCmds(multi []commander) (command, statement string, size int) {
+	var commands []string
+	var statements []string
+	for _, cmd := range multi {
+		cmdStr, stmt, cmdSize := processCmd(cmd)
+		size += cmdSize
+		commands = append(commands, cmdStr)
+		statements = append(statements, stmt)
+	}
+	command = strings.Join(commands, " ")
+	statement = strings.Join(statements, "\n")
+	return command, statement, size
+}
+
+func processMultiCompleted(multi ...valkey.Completed) (command, statement string, size int) {
+	cmds := make([]commander, len(multi))
+	for i, cmd := range multi {
+		cmds[i] = &cmd
+	}
+	return processMultiCmds(cmds)
+}
+
+func processMultiCacheableTTL(multi ...valkey.CacheableTTL) (command, statement string, size int) {
+	cmds := make([]commander, len(multi))
+	for i, cmd := range multi {
+		cmds[i] = &cmd.Cmd
+	}
+	return processMultiCmds(cmds)
+}
+
+func firstError(s []valkey.ValkeyResult) error {
+	for _, result := range s {
+		if err := result.Error(); err != nil && !valkey.IsValkeyNil(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func setClientCacheTags(s tracer.Span, result valkey.ValkeyResult) {
+	s.SetTag(ext.ValkeyClientCacheHit, result.IsCacheHit())
+	s.SetTag(ext.ValkeyClientCacheTTL, result.CacheTTL())
+	s.SetTag(ext.ValkeyClientCachePTTL, result.CachePTTL())
+	s.SetTag(ext.ValkeyClientCachePXAT, result.CachePXAT())
+}
+
+type buildStartSpanOptionsInput struct {
+	command        string
+	statement      string
+	size           int
+	isWrite        bool
+	isBlock        bool
+	isMulti        bool
+	isStream       bool
+	skipRawCommand bool
+}
+
+func (c *coreClient) peerTags() []tracer.StartSpanOption {
```

Comment:
we shouldn't be setting `peer.*` tags in this integration. You can mirror the behavior of the go-redis integration for these tags: https://github.com/DataDog/dd-trace-go/blob/main/contrib/redis/go-redis.v9/redis.go#L75-L111 (changing redis -> valkey where appropiate)

Reply:
> we shouldn't be setting peer.* tags in this integration.

Could you explain the reason for this?

After reading the following docs, I understood `peer.*` is necessary tags now. Since the last commit for go-redis.v9 was two years ago, I assumed that it was not required at that time.

- [DD_APM_PEER_TAGS_AGGREGATION](https://github.com/DataDog/datadog-agent/blob/6c020c8323d53d725bef9f07f0f5a9d06111ffbe/pkg/config/config_template.yaml#L1370-L1381)
- [The new way of overriding service names](https://datadoghq.atlassian.net/wiki/spaces/APST/pages/4132110726/The+new+way+of+overriding+service+names)

I'd appreciate it if you could let me know if my understanding is incorrect.